module-apps/apps-common/widgets/BellSideListItem.hpp

```diff
@@ -22,10 +16,12 @@ namespace gui
         BellBaseLayout *body = nullptr;
 
         explicit BellSideListItem(const std::string &description);
+        BellSideListItem();
```

Comment:
not, becaues we need to call  init() functuin

Reply:
What's the semantic difference between default-constructed item and an item with empty description?

Reply:
I see what is done internally but what does it mean?
Is it simply an item without top message / description or is it something else?
If `topMessage` being null is a valid state then why is it initialized if description is empty?
Why in this case the base class doesn't need a description?
What is the meaning of an `ListItemWithDescription` without description?

Reply:
_"I see what is done internally but what does it mean?"_
It is quite general question but answers for following questions would give you overview.

_"Is it simply an item without top message / description or is it something else?"_
Yes it is simply an item without top message.

_"If topMessage being null is a valid state then why is it initialized if description is empty?"_
Is this question why **BellSideListItem(const std::string &description)** where description is an empty string would not be chceck by if(description.empty())?
Probably because no one have done as detailed analysis as you and naming in this class is a bit off.

_"Why in this case the base class doesn't need a description?"_
Because topMessage(body->first) has to be controlled by the child(DateSetListItem) as the topMerssage has to change "on fly".

_"What is the meaning of an ListItemWithDescription without description?"_
The class ListItemWithDescription introduces the description member but its not used anywhere and its not needed at all. For me its ListItemWithDescription without description its just LIstItem.

Reply:
Inspired by your questions I did remove ListItemWithDescription class entirely from code base as not needed and misleading. BellSideListItem derives form ListItem now. Check now how it looks like.